infrastructure/modules/api-gateway-permissions/locals.tf

```diff
@@ -1,6 +1,5 @@
 # ==============================================================================
 # Context
-
 locals {
-  workspace_suffix = "${terraform.workspace}" == "default" ? "" : "-${terraform.workspace}"
+  workspace_suffix = "${terraform.workspace}" == "default" || "${terraform.workspace}" == "" ? "" : "-${terraform.workspace}"
```

Comment:
Also just for clarificaton - what path to this has terraform.workspace set to "" ? Again just so understand ... I was half-expecting it to be default or something like dr-999 ?

Reply:
terraform.workspace would be set to empty ("") when no explicit workspace has been selected in Terraform. However, this scenario is quite unlikely as Terraform sets a default workspace named 'default' when no explicit workspace is selected.

You're correct that in typical usage you might see terraform.workspace set to something like 'default' or 'dr-999', assuming 'dr-999' is a workspace that has been created. The condition we've specified in this code handles both the default case and any additional workspaces that might have been created.